sqlx-core/src/executor.rs

```diff
@@ -148,10 +149,10 @@ pub trait Executor<'c>: Send + Debug + Sized {
     /// This explicit API is provided to allow access to the statement metadata available after
     /// it prepared but before the first row is returned.
     #[inline]
-    fn prepare<'e, 'q: 'e>(
+    fn prepare<'e>(
         self,
-        query: &'q str,
-    ) -> BoxFuture<'e, Result<<Self::Database as Database>::Statement<'q>, Error>>
+        query: impl SqlSafeStr,
```

Comment:
Maybe just use `SqlStr` here (and on `prepare_with` and `describe`) to avoid monomorphization.